diff --git a/build.sc b/build.sc
index 1831d21..8cacf84 100644
--- a/build.sc
+++ b/build.sc
@@ -36,6 +36,7 @@ trait AutowireModule extends CrossScalaModule with PublishModule{
     ivy"com.lihaoyi::acyclic:${acyclicVersion(scalaVersion())}",
     ivy"org.scala-lang:scala-reflect:${scalaVersion()}"
   )
+  def artifactName = "autowire"
   def publishVersion = "0.3.2"
   def pomSettings = PomSettings(
     description = artifactName(),
